app/repositories/page-content.tsx

```diff
@@ -112,27 +122,42 @@ function useSearchResultsQuery(q: string) {
       return
     }
     setIsAdding(true)
+    const newAbortController = new AbortController()
```

Comment:
For good measure, we should ensure that only one run is possible at any given time.

```suggestion
    // allready running
    if (abortController.current) return

    const newAbortController = new AbortController()
```
or

```suggestion
    // user cancelled by confirming again
    abortController.current?.abort('user-cancelled')
    const newAbortController = new AbortController()
```

Reply:
hmm... not sure how this case could happen since every interaction aborts the controller after it is instantiated. especially with the abort on unmount. I will give it another look when refactoring these into a hook.

Reply:
It shouldn't happen, but only because the `isAdding` UI prevents it to. So if someone else breaks that code (eg. by changing the UI), it might end-up happening in parallel. Adding this is just a safety measure against future changes.